plugins/adrv9002.c

```diff
@@ -927,6 +1189,1009 @@ static void load_profile(GtkFileChooser *chooser, gpointer data)
+	case 2:
+		sprintf(str_value, "LVDS");
+		break;
+	case 4:
+		sprintf(str_value, "CMOS");
+		break;
+	default:
+		profile_gen_set_debug_info(priv, "\nFailed to get ssi_lanes!");
+		sprintf(str_value, "failed to read");
+		break;
+	}
+	gtk_label_set_label(GTK_LABEL(gtk_builder_get_object(priv->builder, "label_radio_ssi")), str_value);
+
+	// duplex mode
+	sprintf(str_value, "%d", cfg->radio_cfg.fdd);
+	gtk_combo_box_set_active_id(GTK_COMBO_BOX(gtk_builder_get_object(priv->builder, "cb_radio_duplex")), str_value);
+
+	// RX
+	for (i = 0; i < 2; i++) {
+		// channel_bandwidth_hz
+		sprintf(widget_str, "cb_rx_chan%d_bw", i + 1);
+		sprintf(value, "%d", cfg->radio_cfg.rx_config[i].channel_bandwidth_hz);
+		populate_combo_box(GTK_COMBO_BOX_TEXT(gtk_builder_get_object(priv->builder, widget_str)), NULL, 0, TRUE, value);
+
+		// sample_rate_hz
+		sprintf(widget_str, "cb_rx_chan%d_interface", i + 1);
+		sprintf(value, "%d", cfg->radio_cfg.rx_config[i].sample_rate_hz);
+		populate_combo_box(GTK_COMBO_BOX_TEXT(gtk_builder_get_object(priv->builder, widget_str)), NULL, 0, TRUE, value);
+
+		// enabled
+		sprintf(widget_str, "cb_rx_chan%d_en", i + 1);
+		if (reset_preset) {
+			gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)), cfg->radio_cfg.rx_config[i].enabled);
+		}
+
+		// frequency_offset_correction_enable
+		sprintf(widget_str, "cb_rx_chan%d_correction", i + 1);
+		if (reset_preset) {
+			gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)), cfg->radio_cfg.rx_config[i].frequency_offset_correction_enable);
+		}
+
+		// rf_port
+		sprintf(widget_str, "cb_rx_chan%d_rf_port", i + 1);
+		sprintf(value, "%d", cfg->radio_cfg.rx_config[i].rf_port);
+		if (reset_preset) {
+			gtk_combo_box_set_active_id(GTK_COMBO_BOX(gtk_builder_get_object(priv->builder, widget_str)), value);
+		}
+	}
+
+	// TX
+	for (i = 0; i < 2; i++) {
+		// channel_bandwidth_hz
+		sprintf(widget_str, "cb_tx_chan%d_bw", i + 1);
+		sprintf(value, "%d", cfg->radio_cfg.tx_config[i].channel_bandwidth_hz);
+		populate_combo_box(GTK_COMBO_BOX_TEXT(gtk_builder_get_object(priv->builder, widget_str)), NULL, 0, TRUE, value);
+
+		// sample_rate_hz
+		sprintf(widget_str, "cb_tx_chan%d_interface", i + 1);
+		sprintf(value, "%d", cfg->radio_cfg.tx_config[i].sample_rate_hz);
+		populate_combo_box(GTK_COMBO_BOX_TEXT(gtk_builder_get_object(priv->builder, widget_str)), NULL, 0, TRUE, value);
+
+		// enabled
+		sprintf(widget_str, "cb_tx_chan%d_en", i + 1);
+		if (reset_preset) {
+			gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)), cfg->radio_cfg.tx_config[i].enabled);
+		}
+
+		// frequency_offset_correction_enable
+		sprintf(widget_str, "cb_tx_chan%d_correction", i + 1);
+		if (reset_preset) {
+			gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)), cfg->radio_cfg.tx_config[i].frequency_offset_correction_enable);
+		}
+	}
```

Comment:
The above two loops look terribly similar. I'm fairly sure you can accommodate them in an helper function